src/Elliptic/DiscontinuousGalerkin/DgOperator.hpp

```diff
@@ -0,0 +1,976 @@
+      ::dg::lift_flux(make_not_null(&auxiliary_boundary_corrections),
+                      mesh.extents(direction.dimension()),
+                      face_normal_magnitude);
+      // The `dg::lift_flux` function contains an extra minus sign
+      auxiliary_boundary_corrections *= -1.;
+
+      // Add the boundary corrections to the auxiliary variables
+      add_slice_to_data(make_not_null(&primal_fluxes_corrected),
+                        auxiliary_boundary_corrections, mesh.extents(),
+                        direction.dimension(), slice_index);
+    }  // apply auxiliary boundary corrections on all mortars
+
+    // Compute the primal equation, i.e. the actual DG operator, by taking the
+    // second derivative: -div(F_u(v)) + S_u = f(x)
+    divergence(operator_applied_to_vars, primal_fluxes_corrected, mesh,
+               inv_jacobian);
+    // This is the sign flip that makes the operator _minus_ the Laplacian for a
+    // Poisson system
+    *operator_applied_to_vars *= -1.;
+    // Using the non-boundary-corrected primal fluxes to compute sources here
+    std::apply(
+        [&operator_applied_to_vars, &primal_vars,
+         &primal_fluxes](const auto&... expanded_sources_args) noexcept {
+          SourcesComputer::apply(
+              make_not_null(&get<OperatorTags>(*operator_applied_to_vars))...,
+              expanded_sources_args..., get<PrimalVars>(primal_vars)...,
+              get<PrimalFluxesVars>(primal_fluxes)...);
+        },
+        sources_args);
+
+    // Add boundary corrections to primal equation
+    for (auto& [mortar_id, mortar_data] : *all_mortar_data) {
+      const auto& [direction, neighbor_id] = mortar_id;
+      const bool is_internal =
+          (neighbor_id != ElementId<Dim>::external_boundary_id());
+      if constexpr (DataIsZero) {
+        if (is_internal) {
+          continue;
+        }
+      }
+      const auto face_mesh = mesh.slice_away(direction.dimension());
+      const size_t slice_index = index_to_slice_at(mesh.extents(), direction);
+      const auto [local_data, remote_data] = mortar_data.extract();
+      const auto& face_normal_magnitude =
+          is_internal ? internal_face_normal_magnitudes.at(direction)
+                      : external_face_normal_magnitudes.at(direction);
+      const auto& mortar_mesh =
+          is_internal ? all_mortar_meshes.at(mortar_id) : face_mesh;
+      const auto& mortar_size =
+          is_internal ? all_mortar_sizes.at(mortar_id) : full_mortar_size;
+
+      // This is the _strong_ primal boundary correction avg(n.F_u) - penalty *
+      // jump(n.F_v) - n.F_u. Note that the "internal penalty" numerical flux
+      // (as opposed to the LLF flux) uses the raw field derivatives without
+      // boundary corrections in the average, which is why we can communicate
+      // the data so early together with the auxiliary boundary data. In this
+      // case the penalty needs to include a factor p^2 / h (see the `penalty`
```

Comment:
the documentation said you use `N_points`, is `p=N_points` here? That would be good to clarify :)